cosmo_tester/test_suites/test_manager_upgrade/manager_upgrade_test.py

```diff
@@ -0,0 +1,311 @@
+            upgrade_blueprint_path, 'simple-manager-blueprint.yaml')
+
+    def upgrade_manager(self, upgrade_blueprint):
+
+        upgrade_inputs = {
+            # TODO: 127.0.0.1? should use openstack output probably
+            # after cfy-5332 gets merged
+            'private_ip': '127.0.0.1',
+            'public_ip': self.upgrade_manager_ip,
+            'ssh_key_filename': self.manager_inputs['ssh_key_filename'],
+            'ssh_user': self.manager_inputs['ssh_user'],
+            'elasticsearch_endpoint_port': 9200
+
+        }
+        upgrade_inputs_file = self.manager_cfy._get_inputs_in_temp_file(
+            upgrade_inputs, self._testMethodName)
+
+        self.manager_cfy.set_maintenance_mode(True)
+        self.manager_cfy.upgrade_manager(
+            blueprint_path=upgrade_blueprint,
+            inputs_file=upgrade_inputs_file)
+
+    def post_upgrade_checks(self, deployment_id):
+        self.rest_client.blueprints.list()
+        self.check_influx(deployment_id)
+
+    def check_influx(self, deployment_id):
+        # TODO influx config should be pulled from props?
+        influx_client = InfluxDBClient(self.upgrade_manager_ip, 8086,
+                                       'root', 'root', 'cloudify')
+        try:
+            result = influx_client.query('select * from /^{0}\./i '
+                                         'where time > now() - 5s'
+                                         .format(deployment_id))
+        except NameError as e:
+            self.fail('monitoring events list for deployment with ID {0} were'
+                      ' not found on influxDB. error is: {1}'
+                      .format(deployment_id, e))
+
+        self.assertTrue(len(result) > 0)
+
+    def rollback_manager(self, rollback_blueprint):
+        rollback_inputs = {
+            'private_ip': '127.0.0.1',
+            'public_ip': self.upgrade_manager_ip,
+            'ssh_key_filename': self.manager_inputs['ssh_key_filename'],
+            'ssh_user': self.manager_inputs['ssh_user'],
+        }
+        rollback_inputs_file = self.manager_cfy._get_inputs_in_temp_file(
+            rollback_inputs, self._testMethodName)
+        self.manager_cfy.set_maintenance_mode(True)
+        self.manager_cfy.rollback_manager(
+            blueprint_path=rollback_blueprint,
+            inputs_file=rollback_inputs_file)
+
+    def post_rollback_checks(self):
+        pass
+
+    def uninstall_deployment(self, deployment_id):
+        self.manager_cfy.executions.start(workflow='uninstall',
+                                          deployment_id=deployment_id).wait()
+
+    def teardown_manager(self):
+        self.manager_cfy.teardown(ignore_deployments=True).wait()
+
+    def _allow_port_on_security_group(self, group_name, port, proto='tcp'):
+        # TODO this is copied from secured_broker_test, probably should
+        # either factor this out, or instead use YamlPatcher to allow the ports
+        # on the blueprint
```

Comment:
hmm, but that is only used for the TestEnvironment blueprint? in here, we're cloning a blueprint without using the TestEnvironment machinery (because here, the manager is part of the system under test, so we don't want to use the testenv manager)